snow/consensus/snowball/binary_snowflake.go

```diff
@@ -32,7 +40,24 @@ type binarySnowflake struct {
 	finalized bool
 }
 
-func (sf *binarySnowflake) RecordSuccessfulPoll(choice int) {
+func (sf *binarySnowflake) RecordPoll(count, choice int) {
+	if sf.finalized {
+		return // This instance is already decided.
+	}
+
+	switch {
+	case count >= sf.alphaConfidence:
+		sf.recordSuccessfulPoll(choice)
+	case count >= sf.alphaPreference:
+		sf.recordPollPreference(choice)
+	default:
+		// If the poll was unsuccessful, RecordUnsuccessfulPoll should
+		// have been called instead.
+		sf.RecordUnsuccessfulPoll()
+	}
```

Comment:
I feel like we should just delete `recordSuccessfulPoll` and `recordPollPreference`.
```suggestion
	if count < sf.alphaPreference {
		// If the poll was unsuccessful, RecordUnsuccessfulPoll should
		// have been called instead.
		sf.RecordUnsuccessfulPoll()
		return
	}
	if count < sf.alphaConfidence {
		sf.confidence = 0
		sf.binarySlush.RecordSuccessfulPoll(choice)
		return
	}

	if preference := sf.Preference(); preference == choice {
		sf.confidence++
	} else {
		// confidence is set to 1 because there has already been 1 successful
		// poll, namely this poll.
		sf.confidence = 1
	}
	sf.finalized = sf.confidence >= sf.beta
	sf.binarySlush.RecordSuccessfulPoll(choice)
```
This comment applies to all the other `snowflake` `RecordPoll` functions as well

Reply:
Will update - I was originally going for making this the minimal possible change